From a79bb936cae9d3e77dbd8727f4860151d3c4ef33 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 9 Oct 2024 11:37:44 +0200
Subject: [PATCH 1281/1695] clk: stm32mp25: fix FLEXGEN configuration of
 pred-division by 1024

The FLEXGEN pre-divider mask (FLEX_PDIV_MASK) is incorrect.
It is not possible to configure it with a div1024 value.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git de64e325bcd32fb54de69b2876fa53b52492f939 clk: stm32mp25: fix FLEXGEN configuration of pred-division by 1024]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I2ad674ed85b3e041ab58b321c4d22f81ff62ac6f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/410061
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/include/dt-bindings/clock/stm32mp25-clksrc.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/core/include/dt-bindings/clock/stm32mp25-clksrc.h b/core/include/dt-bindings/clock/stm32mp25-clksrc.h
index e949ddf80..a02733601 100644
--- a/core/include/dt-bindings/clock/stm32mp25-clksrc.h
+++ b/core/include/dt-bindings/clock/stm32mp25-clksrc.h
@@ -57,14 +57,14 @@
 #define CLK_NOMUX	0x00000010
 
 /* Flexgen define */
-#define FLEX_ID_SHIFT	13
-#define FLEX_SEL_SHIFT	9
+#define FLEX_ID_SHIFT	20
+#define FLEX_SEL_SHIFT	16
 #define FLEX_PDIV_SHIFT	6
 #define FLEX_FDIV_SHIFT	0
 
-#define FLEX_ID_MASK	GENMASK_32(18, 13)
-#define FLEX_SEL_MASK	GENMASK_32(12, 9)
-#define FLEX_PDIV_MASK	GENMASK_32(8, 6)
+#define FLEX_ID_MASK	GENMASK_32(25, 20)
+#define FLEX_SEL_MASK	GENMASK_32(19, 16)
+#define FLEX_PDIV_MASK	GENMASK_32(15, 6)
 #define FLEX_FDIV_MASK	GENMASK_32(5, 0)
 
 #define DIV_CFG(div_id, div)	((CMD_DIV << CMD_SHIFT) |\
-- 
2.47.2

